backend/services/implementations/testService.ts

```diff
@@ -156,9 +156,41 @@ class TestService implements ITestService {
     }
   }
 
+  async publishTest(id: string): Promise<TestResponseDTO> {
+    let test: Test | null;
+
+    try {
+      test = await MgTest.findOneAndUpdate(
+        { _id: id, status: AssessmentStatus.DRAFT },
+        {
+          $set: {
+            status: AssessmentStatus.PUBLISHED,
+          },
+        },
+        {
+          new: true,
+          runValidators: true,
+        },
+      );
+
+      if (!test) {
+        throw new Error(
+          `Test with ID ${id} is not found or not in draft status`,
```

Comment:
thought might be better to query and validate status at the same time, but we lose ability to differentiate the two errors